tests/query/highlights/gleam/function.gleam

```diff
@@ -60,68 +60,68 @@ fn list_of_two(my_value: a) -> List(a) {
 
 fn replace(
 // <- keyword.function
-// ^ function
+// ^^^^^^^ function
 //        ^ punctuation.bracket
   in string: String,
-  // <- symbol
-  // ^ parameter
-  //       ^ parameter
-  //         ^ type
+  // <- label
+  // ^^^^^^ parameter
+  //       ^ punctuation.delimiter
+  //         ^^^^^^ type.builtin
   //               ^ punctuation.delimiter
   each pattern: String,
-  // <- symbol
-  //   ^ parameter
-  //          ^ parameter
-  //            ^ type
+  // <- label
+  //   ^^^^^^^ parameter
+  //          ^ punctuation.delimiter
+  //            ^^^^^^ type.builtin
   //                  ^ punctuation.delimiter
   with replacement: String,
-  // <- symbol
-  //   ^ parameter
-  //              ^ parameter
-  //                ^ type
+  // <- label
+  //   ^^^^^^^^^^^ parameter
+  //              ^ punctuation.delimiter
+  //                ^^^^^^ type.builtin
   //                      ^ punctuation.delimiter
 ) {
   replace(in: "A,B,C", each: ",", with: " ")
-  // <- function
+  // <- function.call
   //     ^ punctuation.bracket
-  //      ^ symbol
-  //        ^ symbol
-  //          ^ string
+  //      ^^ label
+  //        ^ punctuation.delimiter
+  //          ^^^^^^^ string
   //                 ^ punctuation.delimiter
-  //                   ^ symbol
-  //                       ^ symbol
-  //                         ^ string
+  //                   ^^^^ label
+  //                       ^ punctuation.delimiter
+  //                         ^^^ string
   //                            ^ punctuation.delimiter
-  //                              ^ symbol
-  //                                  ^ symbol
-  //                                    ^ string
+  //                              ^^^^ label
+  //                                  ^ punctuation.delimiter
+  //                                    ^^^ string
   //                                       ^ punctuation.bracket
 }
 // <- punctuation.bracket
 
 pub external fn random_float() -> Float = "rand" "uniform"
 // <- type.qualifier
-//  ^ type.qualifier
-//           ^ keyword.function
-//              ^ function
+//  ^^^^^^^^ type.qualifier
+//           ^^ keyword.function
+//              ^^^^^^^^^^^^ function
```

Comment:
The checker crate and our test suite will just check the last caret. Should this become more fine-grained?

Reply:
I'd say yes, it's nicer to cover in case of say builtin types vs regular type identifiers